### PR TITLE
fix: [CDS-73474]: AWS SAM Values.yaml expression resolution is not working

### PR DESCRIPTION
**Pull request by sainath-b**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/50079)

### CODE REVIEW

**Comment by sainath-b** [comment]
trigger uts

**Comment by sainath-b** [comment]
trigger uts

File: 125-cd-nextgen/src/main/java/io/harness/cdng/aws/sam/DownloadManifestsStepHelper.java [hunk 1]
```diff
@@ -111,6 +112,11 @@ public GitCloneStepNode getGitCloneStepNode(
         .build();
   }
 
+  public String getValuesPathFromValuesManifestOutcome(ValuesManifestOutcome valuesManifestOutcome) {
+    GitStoreConfig gitStoreConfig = (GitStoreConfig) valuesManifestOutcome.getStore();
+    return "/harness/" + valuesManifestOutcome.getIdentifier() + "/" + gitStoreConfig.getPaths().getValue().get(0);
```
Comment (abhinav-harness) [inline 125-cd-nextgen/src/main/java/io/harness/cdng/aws/sam/DownloadManifestsStepHelper.java:117]: size check?

**Review by rohitk-h — APPROVED** [review]
> Reviewed 5 of 6 files at r1, 2 of 2 files at r2, 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/50079)* status: 8 file(s) to review. 1 of 2 approvals obtained. 

___
*[`125-cd-nextgen/src/main/java/io/harness/cdng/aws/sam/AwsSamDeployStep.java` line 92 at r3](https://reviewable.io/reviews/harness/harness-core/50079#-NZSpO0N-J4YBebNJPna:-NZSpO0N-J4YBebNJPnb:bmvdcla) ([raw file](https://github.com/harness/harness-core/blob/65ea6202ae6468500946f718079cadf64da5efc7/125-cd-nextgen/src/main/java/io/harness/cdng/aws/sam/AwsSamDeployStep.java#L92)):*
> ```Java
> 
>       awsSamStepHelper.updateServerInstanceInfoList(serverInstanceInfoList, infrastructureOutcome);
>       log.info("Saving AWS SAM server instances into sweeping output");
> ```

nit: we could print number of instances we are going to publish.

File: 125-cd-nextgen/src/main/java/io/harness/cdng/aws/sam/DownloadManifestsStepHelper.java (hunk 1 above)
Comment (sainath-b) [reply 125-cd-nextgen/src/main/java/io/harness/cdng/aws/sam/DownloadManifestsStepHelper.java:117]: Noted this. I will do backend validation as a part of https://harness.atlassian.net/browse/CDS-72430. But we are already having UI validation for this.

**Comment by sainath-b** [comment]
trigger uts

**Comment by sainath-b** [comment]
trigger ut8

**Comment by sainath-b** [comment]
trigger messagecheck

**Review by anilchowdhury-harness — APPROVED** [review]

**Comment by sainath-b** [comment]
trigger review

**Pull request merged**